test/integration/crud/find_cursor_methods.test.js

```diff
@@ -361,4 +361,93 @@ describe('Find Cursor', function () {
       }
     });
   });
+
+  describe('next + Symbol.asyncIterator()', function () {
+    let client;
+    let collection;
+    let cursor;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient();
+      await client.connect();
+      collection = client.db('next-symbolasynciterator').collection('bar');
+      await collection.deleteMany({}, { writeConcern: { w: 'majority' } });
+      await collection.insertMany([{ a: 1 }, { a: 2 }], { writeConcern: { w: 'majority' } });
+    });
+
+    afterEach(async function () {
+      await cursor.close();
+      await client.close();
+    });
+
+    context('when all documents are retrieved in the first batch', function () {
+      it('allows combining iteration modes', async function () {
+        let count = 0;
+        cursor = collection.find().map(doc => {
+          count++;
+          return doc;
+        });
+
+        await cursor.next();
+        // eslint-disable-next-line no-unused-vars
+        for await (const _ of cursor) {
+          /* empty */
+        }
+
+        expect(count).to.equal(2);
+      });
+
+      it('works with next + next() loop', async function () {
+        let count = 0;
+        cursor = collection.find().map(doc => {
+          count++;
+          return doc;
+        });
+
+        await cursor.next();
+
+        let doc;
+        while ((doc = (await cursor.next()) && doc != null)) {
+          /** empty */
+        }
+
+        expect(count).to.equal(2);
+      });
+    });
+
+    context('when there are documents are not retrieved in the first batch', function () {
+      it('allows combining iteration modes', async function () {
```

Comment:
From slack, a few more cases:
- Async iterator first, cursor.next() after.
- Is it our expectation that cursor.next/asyncIterator/tryNext are all UB if someone has made a cursor.stream()? (This was my understanding)
- We have some `readBufferedDocuments()` tests in this file, but not interleaved with next calls, nor in relation to cursor state, what does this "iteration" method do when the cursor is closed? 
- Should we double check tryNext here as well? it doesn't share code paths with next()
- I think after toArray is called the cursor is clearly "done" what do next/asyncIterator/tryNext/readBufferedDocuments/.stream() do after that?